clmm/modbackend/ccl.py

```diff
@@ -97,8 +97,8 @@ def eval_sigma_crit(self, z_len, z_src):
 
         if np.any(z_cut):
             Ds = ccl.angular_diameter_distance(self.cosmo.be_cosmo, a_src[z_cut])
-            Dl = ccl.angular_diameter_distance(self.cosmo.be_cosmo, a_len)
-            Dls = ccl.angular_diameter_distance(self.cosmo.be_cosmo, a_len, a_src[z_cut])
+            Dl = ccl.angular_diameter_distance(self.cosmo.be_cosmo, a_len[z_cut])
+            Dls = ccl.angular_diameter_distance(self.cosmo.be_cosmo, a_len[z_cut], a_src[z_cut])
```

Comment:
This requires the lengths of a_len and a_src are equal. Can we double check that?

Reply:
@shenmingfu Thanks for pointing this potential source of error. I looked into it but actually, the way the modeling functions are vectorized do not support z_cl to be array-like. It should be a float and the docstrings in `clmm_modeling.py` specify that. So I don't think we need to perform an extra check here as `a_len` should be a float, which is then converted into an array of the same length as `a_src` by `a_len = np.repeat(a_len, len(a_src))`.

So I think this is OK and hope this answers your concern.